Tribler/community/tunnel/tunnel_community.py

```diff
@@ -219,7 +219,7 @@ def __init__(self, community):
         self.index = -1
 
     def has_options(self, hops):
-        return len(self.community.active_data_circuits(hops)) > 0
+        return len(self.community.socks_circuits()) > 0
```

Comment:
You can ignore most of this stuff in the PR - it's just crap for debugging,  :+1: 

Primary goal was to get hidden services running, without the need of any exit-enabled nodes. But after setting `min_circuits=0`, the whole downloading over proxies thing stops to work, resulting in strange errors caused by the socks (somehow requiring circuits to exist). I don't know why it requires those circuits and what type of data is sent through them (apparently it sends data over the circuit without using the exit_data() in tunnelcommunity?). According to @egbertbouman Chris was stuck on this for several months, and 'solved' by waiting until there are at least 4 circuits. I'm not sure whether it is worth to solve the underlying problem, so have to accept the fact it requires at least a number of exiting circuits, even when they're not used. Any downloader will create those circuits anyway when downloading files. What do you think about this?